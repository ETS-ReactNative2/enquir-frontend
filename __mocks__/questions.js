const questions = [
    {
        id: 0,
        questionInput: 'If mirrors aren\'t real, how can mirrors be real?',
    },
];

module.exports = {
    questions
};