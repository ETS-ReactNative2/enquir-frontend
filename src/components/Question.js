import React, { Component } from 'react';
import { Modal, Text, FlatList, Button, View, StyleSheet, TextInput, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

export default class Question extends Component {
    state = {
        question: this.props.question,
        editing: false,
    };

    edit = () => {
        this.setState({
            editing: !this.state.editing,
        });
    };

    render() {
        const { question, editing } = this.state;

        return (
            <View style={styles.questionContainer}>
                <TextInput
                    style={styles.questionText}
                    value={question.item}
                    editable={editing}
                />
                <View style={styles.questionEditContainer}>
                    <TouchableOpacity onPress={this.edit}>
                        <View style={styles.editQuestion}>
                            <Icon name='pencil' size={20} />
                        </View>
                    </TouchableOpacity>
                    <TouchableOpacity onPress={() => this.props.delete(question)}>
                        <View style={styles.deleteQuestion}>
                            <Icon name='trash' size={20} />
                        </View>
                    </TouchableOpacity>
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    questionContainer: {
        flexDirection: 'row',  // main axis
        justifyContent: 'space-between', // main axis
        paddingTop: 10,
        paddingBottom: 10,
        paddingLeft: 18,
        paddingRight: 16,
        marginTop: 0,
        marginBottom: 8,
        elevation: 1,
        borderRadius: 3,
        backgroundColor: 'grey',
    },
    questionText: {
        justifyContent: 'flex-start',
    },
    questionEditContainer: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'flex-end',
    },
    editQuestion: {
        marginRight: 12,
    },
    deleteQuestion: {},
});