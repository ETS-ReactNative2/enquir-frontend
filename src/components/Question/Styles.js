import { StyleSheet } from 'react-native';

const containers = StyleSheet.create({
    questionContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        paddingTop: 10,
        paddingBottom: 10,
        paddingLeft: 18,
        paddingRight: 16,
        marginTop: 0,
        marginBottom: 8,
        elevation: 1,
        borderRadius: 3,
        backgroundColor: 'grey',
    },
    questionTextContainer: {
        flex: 12,
        flexDirection: 'row',
        justifyContent: 'flex-start',
    },
    editMenu: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'flex-end',
    },
});

const text = StyleSheet.create({
    questionText: {
        flex: 1,
        flexWrap: 'wrap',
        fontSize: 16,
        fontFamily: 'Roboto-Regular',
    },
});

export { containers, text };