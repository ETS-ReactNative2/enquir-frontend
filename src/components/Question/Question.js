import React, { Component } from 'react';
import { 
    View, 
    TextInput, 
    TouchableOpacity,
} from 'react-native';
import { containers, text } from './Styles';
import Icon from 'react-native-vector-icons/FontAwesome';

export default class Question extends Component {
    state = {
        question: this.props.question,
        questionInput: this.props.question.item,
    };

    render() {
        const { question, questionInput } = this.state;

        return (
            <View style={containers.questionContainer}>
                <View style={containers.questionTextContainer}>
                    <TextInput
                        style={text.questionText}
                        onChangeText={(questionInput) => this.setState({ questionInput })}
                        onEndEditing={() => this.props.save(question.index, questionInput)}
                        value={questionInput}
                        multiline={true}
                    />
                </View>
                <View style={containers.editMenu}>
                    <TouchableOpacity onPress={() => this.props.delete(question.index)}>
                        <View>
                            <Icon name='trash' size={18} />
                        </View>
                    </TouchableOpacity>
                </View>
            </View>
        );
    }
};