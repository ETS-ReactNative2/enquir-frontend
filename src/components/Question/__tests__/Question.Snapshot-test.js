import React from 'react';
import renderer from 'react-test-renderer';

import Question from '../Question';
import { questions } from '../../../../__mocks__/questions';

test('renders a Question using Snapshots', () => {
    expect(renderer.create(
        <Question
            id={questions[0].id}
            questionInput={questions[0].questionInput}
        />
    )).toMatchSnapshot();
});