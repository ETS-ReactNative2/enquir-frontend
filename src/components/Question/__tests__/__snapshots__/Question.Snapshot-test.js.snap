// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Question using Snapshots 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#c3c4c9",
      "borderRadius": 3,
      "elevation": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 8,
      "marginTop": 0,
      "paddingBottom": 10,
      "paddingLeft": 18,
      "paddingRight": 16,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 12,
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
          "flexWrap": "wrap",
          "fontFamily": "Roboto-Regular",
          "fontSize": 18,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 18,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ï‡¸
        </Text>
      </View>
    </View>
  </View>
</View>
`;
